hudi-common/src/main/java/org/apache/hudi/common/table/read/UnmergedFileGroupRecordBuffer.java

```diff
@@ -69,19 +71,31 @@ protected boolean doHasNext() {
     while ((recordIterator == null || !recordIterator.hasNext()) && !currentInstantLogBlocks.isEmpty()) {
       HoodieLogBlock logBlock = currentInstantLogBlocks.pop();
       if (logBlock instanceof HoodieDataBlock) {
+        isDeleteBlock = false;
```

Comment:
can we make it local.

Reply:
nope, you can't make this local. it has to retain values across hasNext() calls.

Reply:
Yes it is used here but you need to know that the current iterator is over deletes vs data blocks so you need to maintain that state between calls to `doHasNext`